app/controllers/observations_controller/index.rb

```diff
@@ -114,12 +114,22 @@ def render_observation_search_results(pattern)
       search.errors.each do |error|
         flash_error(error.to_s)
       end
-      render("index", location: observations_path)
+      if params[:needs_id]
+        redirect_to({ controller: "/observations/identify", action: :index,
+                      q: get_query_param })
+      else
+        render("index", location: observations_path)
+      end
     else
       @suggest_alternate_spellings = search.query.params[:pattern]
-      show_selected_observations(
-        search.query, no_hits_title: :title_for_observation_search.t
-      )
+      if params[:needs_id]
+        redirect_to({ controller: "/observations/identify", action: :index,
+                      q: get_query_param })
+      else
```

Comment:
@pellaea I'm having an issue here. 

Calling the identify index with `q: get_query_param` doesn't pull the query we just made, so it calls the identify index without the recent search query.

At the top of this method:
```ruby
  def render_observation_search_results(pattern)
    search = PatternSearch::Observation.new(pattern)
```
Does instantiating `PatternSearch::Observation.new(pattern)` save the query record? Or not yet?
I think i need to save the record before redirecting.

Reply:
Ah. `show_selected_observations` does `store_query_in_session`. That's what i need to do here.

Reply:
Actually i can do the same thing via `get_query_param(search.query)` - it also saves it.